src/relentless/simulate/hoomd.py

```diff
@@ -79,7 +85,22 @@ def _call_v3(self, sim):
             )
             pair_potential.r_cut[(i, j)] = r[-1]
         sim[self]["_potentials"] = [pair_potential]
-        sim[self]["_potentials_rmax"] = r[-1]
+
+        sim[self]["_bonds"] = self._get_bonds_from_snapshot(sim, snap)
+        if snap.bonds.N > 0:
```

Comment:
I think these will cause problems if you try to run the simulation under MPI. The reason is that the snapshot is only valid on the root rank, and your `_get_bonds_from_snapshot` method does not broadcast.

Use `state.N_bonds` if you want access to the number of bonds on all ranks.

For the snapshot method, I would have `_get_bonds_from_snapshot` return `None` on non-root ranks, as I think only the root rank actually needs it later on to process the RDF, right?

Also a small conceptual question: if there are no bonds in the configuration but the user does specify bond potentials, should we attach them anyway? Is performance the reason not to do so?

Reply:
I think performance would be the reason not to, but I don't know how much of an issue that would cause. If we did that, would just a try block work? This will throw an error if `sim.potentials.bond` attribute isn't on Potentials.

Reply:
`Potentials` should have a `bond` property that defaults to `None`, like is done for `pair`. The setter should do type validation on the tabulator as well. Once that's in place, checking for the existence of bond potentials is `sim.potentials.bond is not None`.